### PR TITLE
Starting to refactor SocialGraphPlugin

### PR DESCRIPTION
**Pull request by gmethvin**
I'm trying to move facebook-specific stuff into FacebookSocialGraph to make it easier to add other social networks. Eventually I think we'll have a FacebookSocialGraph, LinkedInSocialGraph, etc. Those are the only classes that should be concerned with the specific JSON returned by the social network.

### CODE REVIEW

File: shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala [hunk 1]
```diff
@@ -1,51 +1,28 @@
...
-          case Some(email) =>
-            if (email.userId != userId) throw new IllegalStateException("email %s is not associated with user %s".format(email, userRepo.get(userId)))
-            log.info("email %s for user %s already exist".format(email, userRepo.get(userId)))
-            email
-          case None =>
-            log.info("creating new email %s for user %s already exist".format(emailString, userRepo.get(userId)))
-            emailRepo.save(EmailAddress(userId = userId, address = emailString))
-        }
+  private def importEmailFromJson(userId: Id[User], emailString: String): EmailAddress = {
+    db.readWrite{ implicit session =>
```
Comment (eishay) [inline shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala]: Correct me if I'm wrong, but it seems like the common use case (social graph refresh) is that there is a hit (i.e. `some(email)`).
Only in cases that users first log in then there's a miss (None). Would it make sense to have `getByAddressOpt` in readOnly then?
note: I think its my own old bad code I'm commenting on there :-)

File: shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala [hunk 2]
```diff
@@ -1,51 +1,28 @@
...
+      emailRepo.getByAddressOpt(emailString) match {
+        case Some(email) =>
+          if (email.userId != userId) throw new IllegalStateException("email %s is not associated with user %s".format(email, userRepo.get(userId)))
+          log.info("email %s for user %s already exist".format(email, userRepo.get(userId)))
```
Comment (eishay) [inline shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala]: [my old bad code] doing a `userRepo.get(userId)` (i.e. going to the db) only for log no one will read, maybe we don't need that.

File: shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala (hunk 1 above)
Comment (gmethvin) [inline shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala]: Yes, I think you're right.

File: shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala (hunk 2 above)
Comment (gmethvin) [inline shoebox/modules/common/app/com/keepit/common/social/SocialUserImportEmail.scala]: Cool. I'll just log the userId.

**Comment by eishay** [comment]
cool

**Pull request merged**